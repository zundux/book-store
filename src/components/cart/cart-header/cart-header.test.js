import React from "react";
import CartHeader from "./cart-header";

describe("Cart grid header", () => {

  test("Cart header has only the text \"Cart\" when cart is empty", () => {
    const component = shallow(
      <CartHeader products={[]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"product\" when cart has only 1 product", () => {
    const component = shallow(
      <CartHeader products={[
        {
          id: 1,
          amount: 1
        }
      ]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"products\" when cart has a couple of products", () => {
    const component = shallow(
      <CartHeader products={[
        {
          id: 1,
          amount: 1
        },
        {
          id: 2,
          amount: 1
        }
      ]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"products\" when cart contain 2 copies of the product", () => {
    const component = shallow(
      <CartHeader products={[
        {
          id: 1,
          amount: 2
        }
      ]}/>
    );
    matchSnapshot(component);
  });

});

