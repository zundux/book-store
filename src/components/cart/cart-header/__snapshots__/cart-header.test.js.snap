// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart grid header Cart header has only the text "Cart" when cart is empty 1`] = `
<div>
  <h1>
    Cart
  </h1>
</div>
`;

exports[`Cart grid header Header has the word "product" when cart has only 1 product 1`] = `
<div>
  <h1>
    Cart
  </h1>
  <div>
    1
     product
     in cart
  </div>
</div>
`;

exports[`Cart grid header Header has the word "products" when cart contain 2 copies of the product 1`] = `
<div>
  <h1>
    Cart
  </h1>
  <div>
    2
     product
    s
     in cart
  </div>
</div>
`;

exports[`Cart grid header Header has the word "products" when cart has a couple of products 1`] = `
<div>
  <h1>
    Cart
  </h1>
  <div>
    2
     product
    s
     in cart
  </div>
</div>
`;
